### PR TITLE
Add Sprint report for Sprint 39

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by hennevogel** [comment]
Deploy preview for *openbuildservice* ready!

Built with commit dbd7ff887639467ad3a736b9f92c187d2d6b1e4d

https://deploy-preview-163--openbuildservice.netlify.com

**Comment by bgeuken** [comment]
@openSUSE/open-build-service Please review

Would be awesome if you could add some summary of the oSC conference

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+category: development
+---
+
+People of the Builds! Another Sprint is over and here is what the OBS frontend team has achieved in the last two weeks (2018-05-21 to 2018-06-01).
+
+# Bugfixes:
+
+## Patchinfo editor removes data
+
+As it was reported in [#4895](https://github.com/openSUSE/open-build-service/issues/4895) after editing a patchinfo with an optional `message` element, the element was removed.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: I think there is a coma missing here: _As it was reported in #4895, after..._

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+This was fixed in [#5010](https://github.com/openSUSE/open-build-service/pull/5010) (screenshots included there).
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: what about adding the screenshots to the blog post? :thinking:

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Working on this issue lead us to [add the missing optional version field](https://github.com/openSUSE/open-build-service/pull/5015), and fix [another issue](https://github.com/openSUSE/open-build-service/pull/5017) related with the selection of binaries.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: no coma needed before and

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+That's the thing fixing issues in OBS: you don't stop when fixing a bug. You dig into the code and make it as good as you can. :sweat_smile:
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: :rofl:

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ 
+## Build result page took long time to load:
+
+In the binaries page we were always asking to the backend per file if it knows the url for downloading.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: I would remove the always as you already say it is per file :thinking:

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+If the backend answers with an url, that url will be checked on the fly to know if the file is available, if it is not there then we provide an API url.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: **past**:

If the backend answered with an url, that url was checked...

An a `.` before the _if it is not there_. Or even better _Otherwise_

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+That meant that up to 2 http request are performed to only know if we need to show the link.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: up to 2? :thinking:

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+I we have a package with 20 files per subpackage and 5 subpackages... we will perform up to 200 requests just to show links.
+Now we have deferred this calculation to an internal action in the frontend to only perform the request to external sources if the user really wants to download a file.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: I think we need a festive emoji here... :tada:

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+# Code Maintenance
+
+## Fix MySQL Collate issues for SLE15
+
+As you might know OBS is using structure.sql for dumping the database schema of our app. :neckbeard:
+This is slightly different from the schema.rb dumps that Rails applications use by default.
+In a nutshell the difference between those two variants is that the structure.sql dumps include database specific settings like triggers, sequences and alike.
+If you want to know more about schema dumps in Rails, I recommend the awesome [Rails guide](http://guides.rubyonrails.org/active_record_migrations.html#schema-dumping-and-you).
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: we instead of I? :thinking:

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+In addition to that we have a fancy migration test that ensures that the structure.sql file we store in our git repository matches with the one created from scratch (based on an OBS 2.5 dump). :sweat:
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: Comma missed: _In addition to that, we..._

File: _posts/development/2018-06-01-sprint-report-39.md (hunk 8 above)
Comment (Ana06) [reply _posts/development/2018-06-01-sprint-report-39.md]: and I would link the PR

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+This test runs within our [RPM package builds](https://build.opensuse.org/package/show/OBS:Server:Unstable/obs-server) for any target we build for.
+
+Now with the upcoming SUSE Linux Enterprise 15 release we started to build packages for this platform and noticed that these tests fail because of a mismatching sql dump. :bomb:
+The reason for that was that newer MySQL / MariaDB versions, like the one we ship with SLE15, introduced a few changes:
+
+* "Numbers are no longer quoted in the DEFAULT clause in SHOW CREATE statement", see [PR#4999](https://github.com/openSUSE/open-build-service/pull/4999)
+* "If no specific default is provided, SHOW CREATE TABLE will append a DEFAULT NULL to nullable TEXT or BLOB fields", see [PR](https://github.com/openSUSE/open-build-service/pull/5001)
+* The default MySQL character set and collation changed from utf8 to utf8mb4, see [PR#5020](https://github.com/openSUSE/open-build-service/pull/5020) and [MySQL](https://dev.mysql.com/doc/refman/8.0/en/charset-applications.html) documentation
+
+As you can see from the pull requests we've added normalization code to our migration code to address the first 2 changes.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: From the PR? I would say in the PR... :thinking:

File: _posts/development/2018-06-01-sprint-report-39.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+So it was time to give OBS an update as well :wink:
+
+Once we've reviewed the changes the Rails folks been doing we've updated https://build.opensuse.org.
+Everything works like a charm and we are now running on Rails 5.2 :heart:
+    
+# Features:
+
+## OBS in Semantic-UI
+
+After trying out [Bulma](https://bulma.io) [last sprint](https://openbuildservice.org/2018/05/23/sprint-report-38) we gave [Semantic-UI](https://semantic-ui.com) a shot.
```
Comment (Ana06) [inline _posts/development/2018-06-01-sprint-report-39.md]: Comma missed after _last sprint_ :thinking:

**Pull request merged**